zerver/models.py

```diff
@@ -2640,33 +2640,29 @@ class UserTopic(models.Model):
         default=datetime.datetime(2020, 1, 1, 0, 0, tzinfo=datetime.timezone.utc)
     )
 
-    # Implicitly, if a UserTopic does not exist, the (user, topic)
-    # pair should have normal behavior for that (user, stream) pair.
+    class VisibilityPolicy(models.IntegerChoices):
```

Comment:
Does every `user_topic` object have a `.VisibilityPolicy`, or is this element only on the `UserTopic` type? (Mostly thinking about the resource usage tradeoffs for this class being defined here vs. at the top-level, and I think the answer is that it doesn't matter, because individual `UserTopic` objects won't have a `.VisibilityPolicy`? But seems worth thinking about when trying to pick these patterns).

Reply:
Individual `UserTopic`s also have the `.VisibilityPolicy` key present in them.

Reply:
I'm not sure there will be noticeable resource usage difference for this table in particular, but if we're thinking of using this pattern throughout the codebase, maybe it would be better to have the choices class outside the main class. Thoughts?

Reply:
I suspect I was just confused above, and it's just a pointer that appears in every `UserTopic` element -- we don't create a new copy of this class for each `UserTopic` object, they just each get a pointer to the single `class UserTopic`. So I think this is fine but maybe let's verify this theory and then post in the #backend thread about it.